docs/book/src/development/tilt-setup.md

```diff
@@ -80,7 +85,70 @@ Next, create a `tilt-settings.json` file and place it in your local copy of `clu
   }
 ```
 
-### Run Tilt!
+### Debugging
+
+If you would like to debug CAPA (or core CAPI / another provider) you can run the provider with delve. This will then allow you to attach to delve and debug.
+
+To do this you need to use the **debug** configuration in **tilt-settings.json**. Full details of the options can be seen [here](https://cluster-api.sigs.k8s.io/developer/tilt.html).
+
+An example **tilt-settings.json**:
+
+```json
+{
+  "enable_providers": [
+    "kubeadm-bootstrap",
+    "kubeadm-control-plane",
+    "aws"
+  ],
+  "default_registry": "gcr.io/your-project-name-here",
+  "provider_repos": [
+    "/Users/username/go/src/sigs.k8s.io/cluster-api-provider-aws"
+  ],
+  "kustomize_substitutions": {
+    "EXP_CLUSTER_RESOURCE_SET": "true",
+    "EXP_MACHINE_POOL": "true",
+    "EVENT_BRIDGE_INSTANCE_STATE": "true",
+    "AWS_B64ENCODED_CREDENTIALS": "W2RlZmFZSZnRg==",
+    "EXP_EKS_FARGATE": "false",
+    "CAPA_EKS_IAM": "false",
+    "CAPA_EKS_ADD_ROLES": "false"
+  },
+  "extra_args": {
+    "aws": ["--v=2"]
+  }
+  "debug": {
+    "aws": {
+      "continue": true,
+      "port": 30000
+    }
+  }
+}
+```
+
+Once you have run tile (see section below) you will be able to connect to the running instance of delve. To this you can use the a VSCode launch configuration like this:
+
+```json
+    {
+        "name": "Connect to CAPA",
+        "type": "go",
+        "request": "attach",
+        "mode": "remote",
+        "remotePath": "",
+        "port": 30000,
+        "host": "127.0.0.1",
+        "showLog": true,
+        "trace": "log",
+        "logOutput": "rpc"
+    }
+```
+
+Or you could use delve directly from the CLI using a command similar to this:
```

Comment:
It would be good to add Intellij too as in [CAPI docs:](https://cluster-api.sigs.k8s.io/developer/tilt.html)
```
Goland / Intellij
With the above example, you can configure a Go Remote run/debug configuration pointing at port 30000.
```

Reply:
Good point, thanks @sedefsavas . I have added this to the doc.